src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -200,29 +200,53 @@ protected override Conversion GetCollectionExpressionConversion(
                 }
             }
 
+            bool usesKeyValuePairs = CollectionUsesKeyValuePairs(Compilation, collectionTypeKind, elementType, out var keyType, out var valueType);
             var builder = ArrayBuilder<Conversion>.GetInstance(elements.Length);
             foreach (var element in elements)
             {
-                Conversion elementConversion = convertElement(element, elementType, ref useSiteInfo);
-                if (!elementConversion.Exists)
+                switch (element)
                 {
-                    builder.Free();
-                    return Conversion.NoConversion;
+                    case BoundCollectionExpressionSpreadElement spreadElement:
+                        {
+                            var elementConversion = GetCollectionExpressionSpreadElementConversion(spreadElement, elementType, ref useSiteInfo);
+                            if (elementConversion.Exists)
+                            {
+                                builder.Add(elementConversion);
+                                continue;
+                            }
+                        }
+                        break;
+                    case BoundKeyValuePairElement keyValuePairElement:
+                        {
+                            if (usesKeyValuePairs)
+                            {
+                                var keyConversion = ClassifyImplicitConversionFromExpression(keyValuePairElement.Key, keyType.Type, ref useSiteInfo);
+                                var valueConversion = ClassifyImplicitConversionFromExpression(keyValuePairElement.Value, valueType.Type, ref useSiteInfo);
+                                if (keyConversion.Exists && valueConversion.Exists)
+                                {
+                                    builder.Add(keyConversion);
+                                    builder.Add(valueConversion);
+                                    continue;
+                                }
+                            }
+                        }
+                        break;
+                    default:
+                        {
+                            var elementConversion = ClassifyImplicitConversionFromExpression((BoundExpression)element, elementType, ref useSiteInfo);
+                            if (elementConversion.Exists)
+                            {
+                                builder.Add(elementConversion);
+                                continue;
+                            }
+                        }
+                        break;
                 }
```

Comment:
nit: `CreateCollectionExpressionConversion` used below takes "elementConversions", but we're storing key and value conversions in there too. Consider renaming the parameter to something more generic, such as "nestedConversions". This would likely flow into `CollectionExpressionUncommonData` too.

Reply:
I'll leave this as "elementConversions" for now since we might need to handle key-value pair conversions as grouped `Conversion` instances.